### DISCUSSION TITLE
Experimental backend question.

### DISCUSSION THREAD

**Discussion by DrTodd13** [dask/dask#10372] [Q&A]
I have been experimenting with ArrayBackendEntrypoint to provide an alternative backing store instead of NumPy.  I have arrays A and B that I create directly through dask and internally those seem to use my configured backing store.  I then do:

B += da.overlap.overlap(A, depth=r, boundary='none').map_blocks(lambda x: stencil(x,W,pattern,r),chunks=csB)

Inside the stencil function, the type of A is numpy.ndarray instead of my alternate backing store like I expected.  Can someone help me understand what is going on in this process to I presume at some point to have converted from my alternate backing store back to NumPy?

@rjzamora

**Comment by rjzamora** [comment]
Sorry for missing this earlier @DrTodd13 !

The `BackendEntrypoint` API is still very experimental, and defining a custom backend is not recommended. The challenge is that there are effectively several dispatching systems that need to be considered and coordinated: (1) the collection-creation API, (1) type-based compute dispatch, and (3) the PEP18  `__array_function__` protocol.

That said, if you still want to experiment with a custom backend for fun/learning, I'll be happy to help :)

My initial guess is that some numpy-specific logic is being used somewhere in `overlap`. However, I would probably need a reproducer to say for sure what is happening.

**Comment by danilopeixoto** [comment]
Hi @rjzamora,
Hope you are doing well!

I'm currently working on a prototype for an array backend. I was wondering if it's feasible to create a backend for array implementation where the data type is not treated as an alias of `np.dtype`?
